projects/element-ng/toast-notification/si-toast-notification.service.ts

```diff
@@ -31,6 +31,11 @@ export class SiToastNotificationService implements OnDestroy {
 
   private injector = inject(Injector);
   private overlay = inject(Overlay);
+  private toastTimeoutMap = new Map<string, any>();
+  private toastTimerDefaults = new Map<
+    string,
```

Comment:
here, no need for ID, just use the `SiToast`

```suggestion
  private toastTimeoutMap = new Map<SiToast, any>();
  private toastTimerDefaults = new Map<
    SiToast,
```